packages/browser/src/integrations/featureFlags/statsig/integration.ts

```diff
@@ -0,0 +1,44 @@
+import type { Client, Event, EventHint, IntegrationFn } from '@sentry/core';
+
+import { defineIntegration } from '@sentry/core';
+import { copyFlagsFromScopeToEvent, insertFlagToScope } from '../../../utils/featureFlags';
+import type { FeatureGate, StatsigClient } from './types';
+
+/**
+ * Sentry integration for capturing feature flag evaluations from the Statsig js-client SDK.
+ *
+ * See the [feature flag documentation](https://develop.sentry.dev/sdk/expected-features/#feature-flags) for more information.
+ *
+ * @example
+ * ```
+ * import { StatsigClient } from '';
+ * import * as Sentry from '@sentry/browser';
+ *
+ * const statsigClient = new StatsigClient();
+ *
+ * Sentry.init({
+ *   dsn: '___PUBLIC_DSN___',
+ *   integrations: [Sentry.statsigIntegration({statsigClient})],
+ * });
+ *
+ * await statsigClient.initializeAsync();  // or statsigClient.initializeSync();
+ *
+ * const result = statsigClient.checkGate('my-feature-gate');
+ * Sentry.captureException(new Error('something went wrong'));
+ * ```
+ */
+export const statsigIntegration = defineIntegration(({ statsigClient }: { statsigClient: StatsigClient }) => {
```

Comment:
Damn we probably should have called this something more generic like `client` so it's the same across integrations

```suggestion
export const statsigIntegration = defineIntegration(({ client }: { client: StatsigClient }) => {
```

Reply:
Wanna start now? Though personally I like being explicit here, having the provider in the name makes it clear what type it excepts (e.g. not Sentry client)

Reply:
yeah `client` not the best choice either bc of what you said

Reply:
deciding on `featureFlagClient`